src/starters/wms-extensions/src/main/java/org/geoserver/cloud/autoconfigure/wms/extensions/GraticuleConfiguration.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) 2024 Open Source Geospatial Foundation - all rights reserved This code is licensed under the
+ * GPL 2.0 license, available at the root application directory.
+ */
+
+package org.geoserver.cloud.autoconfigure.wms.extensions;
+
+import org.geoserver.cloud.config.factory.FilteringXmlBeanDefinitionReader;
+import org.geoserver.platform.ModuleStatus;
+import org.geoserver.platform.ModuleStatusImpl;
+import org.geotools.data.graticule.GraticuleDataStoreFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportResource;
+
+@Configuration
+public class GraticuleConfiguration {
+
+    @Bean
+    @ConditionalOnProperty(
+            prefix = "geoserver.wms.graticule",
+            name = "enabled",
+            havingValue = "true",
+            matchIfMissing = true)
+    public GraticuleDataStoreFactory graticuleDataStoreFactory() {
+        return new GraticuleDataStoreFactory();
+    }
```

Comment:
this is not necessary, `DataStoreFactory`'s are discovered using the Java SPI mechanism (i.e. DataAccessFinder in geotools)